### ISSUE TITLE
nengo.dists.Choice([1]) behaviour can be improved

### ISSUE THREAD

**Issue by arvoelke** [nengo/nengo#1473]
If we create an ensemble such as:

```python
    nengo.Ensemble(100, 1, encoders=np.ones(100))
```

then we get a good error message:

```
ValidationError: Ensemble.encoders: ndarray must be 2D (got 1D)
```

However, as a `Choice` distribution:

```python
with nengo.Network() as model:
    nengo.Ensemble(100, 1, encoders=nengo.dists.Choice([1]))
with nengo.Simulator(model) as sim:
    pass
```

```
~/CTN/nengo/nengo/simulator.py in __init__(self, network, dt, seed, model, progress_bar, optimize)
    158                 # Build the network into the model
    159                 self.model.build(network,
--> 160                                  progress=pt.next_stage("Building", "Build"))
    161 
    162             # Order the steps (they are made in `Simulator.reset`)

~/CTN/nengo/nengo/builder/builder.py in build(self, obj, *args, **kwargs)
    121             The object to build into this model.
    122         """
--> 123         built = self.builder.build(self, obj, *args, **kwargs)
    124         if self.build_callback is not None:
    125             self.build_callback(obj)

~/CTN/nengo/nengo/builder/builder.py in build(cls, model, obj, *args, **kwargs)
    216                 "Cannot build object of type %r" % type(obj).__name__)
    217 
--> 218         return cls.builders[obj_cls](model, obj, *args, **kwargs)
    219 
    220     @classmethod

~/CTN/nengo/nengo/builder/network.py in build_network(model, network, progress)
     95         logger.debug("Network step 1: Building ensembles and nodes")
     96         for obj in network.ensembles + network.nodes:
---> 97             model.build(obj)
     98 
     99         logger.debug("Network step 2: Building subnetworks")

~/CTN/nengo/nengo/builder/builder.py in build(self, obj, *args, **kwargs)
    121             The object to build into this model.
    122         """
--> 123         built = self.builder.build(self, obj, *args, **kwargs)
    124         if self.build_callback is not None:
    125             self.build_callback(obj)

~/CTN/nengo/nengo/builder/builder.py in build(cls, model, obj, *args, **kwargs)
    216                 "Cannot build object of type %r" % type(obj).__name__)
    217 
--> 218         return cls.builders[obj_cls](model, obj, *args, **kwargs)
    219 
    220     @classmethod

~/CTN/nengo/nengo/builder/ensemble.py in build_ensemble(model, ens)
    170         encoders = npext.array(ens.encoders, min_dims=2, dtype=np.float64)
    171     if ens.normalize_encoders:
--> 172         encoders /= npext.norm(encoders, axis=1, keepdims=True)
    173 
    174     # Build the neurons

~/CTN/nengo/nengo/utils/numpy.py in norm(x, axis, keepdims)
    139         newer versions of Numpy (>= 1.7).
    140     """
--> 141     return np.sqrt(np.sum(x**2, axis=axis, keepdims=keepdims))
    142 
    143 

~/anaconda3/envs/py36/lib/python3.6/site-packages/numpy/core/fromnumeric.py in sum(a, axis, dtype, out, keepdims, initial)
   1928 
   1929     return _wrapreduction(a, np.add, 'sum', axis, dtype, out, keepdims=keepdims,
-> 1930                           initial=initial)
   1931 
   1932 

~/anaconda3/envs/py36/lib/python3.6/site-packages/numpy/core/fromnumeric.py in _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs)
     81                 return reduction(axis=axis, out=out, **passkwargs)
     82 
---> 83     return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
     84 
     85 

AxisError: axis 1 is out of bounds for array of dimension 1
```

The correct thing to use is `nengo.dists.Choice([[1]])`.

I think that `Choice` should either throw an error if the input's not 2D, or it should reinterpret it as 2D to be consistent with how `encoders` are given as arrays (and then document + unit-test that behaviour).

**Comment by tbekolay** [comment]
Hm, I thought we did that check [here](https://github.com/nengo/nengo/blob/master/nengo/dists.py#L321). Why is that not working in this case?

**Comment by arvoelke** [comment]
> Hm, I thought we did that check [here](https://github.com/nengo/nengo/blob/master/nengo/dists.py#L321). Why is that not working in this case?

`nengo.dists.Choice([1]).dimensions => 1.0`, and so that check goes through okay. This also raises the point that we should convert `dimensions` to an `int` from a `float` (think this is only the case when `options.shape[1:] => ()`).

**Comment by hunse** [comment]
Fixed by #1630
